Equilobe.DailyReport.SL/AtlassianUserDataService.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Equilobe.DailyReport.DAL;
+using Equilobe.DailyReport.Models.Dashboard;
+using Equilobe.DailyReport.Models.Interfaces;
+using Equilobe.DailyReport.Models.Storage;
+using Equilobe.DailyReport.Utils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Equilobe.DailyReport.SL
+{
+    public class AtlassianUserDataService : IAtlassianUserDataService
+    {
+        public IJiraService JiraService { get; set; }
+        public IAtlassianWorklogDataService AtlassianWorklogDataService { get; set; }
+
+        #region IAtlassianDataService implementation
+        public List<AtlassianUser> GetAtlassianUsers(long instanceId, bool? isActive = null, bool? isStalling = null)
+        {
+            using (var db = new ReportsDb())
+            {
+                return db.AtlassianUsers
+                    .Where(p => p.InstalledInstanceId == instanceId)
+                    .Where(p => !isActive.HasValue || p.IsActive == isActive)
+                    .Where(p => !isStalling.HasValue || p.IsStalling == isStalling)
+                    .ToList();
+            }
+        }
+
+        public void SyncAtlassianUsers(List<AtlassianUser> users, ReportContext context)
+        {
+            UpdateAtlassianUsers(context.InstanceId, users);
+            UpdateUsersAvatars(users, context);
+            UpdateStallingUsers(context);
+        }
+        #endregion
+
+        #region Update methods
+        private void UpdateAtlassianUsers(long instanceId, List<AtlassianUser> users)
+        {
+            using (var db = new ReportsDb())
+            {
+                var dbUsers = db.AtlassianUsers
+                    .Where(p => p.InstalledInstanceId == instanceId);
+
+                foreach (var user in users)
+                {
+                    var dbUser = dbUsers.Where(p => p.Key == user.Key).SingleOrDefault();
+
+                    if (dbUser == null)
+                        db.AtlassianUsers.Add(user);
+                    else
+                        UpdateDbUser(dbUser, user);
+                }
+
+                db.SaveChanges();
+            }
+        }
+
+        private void UpdateUsersAvatars(List<AtlassianUser> users, ReportContext context)
+        {
+            var fileNames = UploadUsersAvatarsAndGetFilenames(users, context);
+
+            UpdateDbUsersAvatars(fileNames, context.InstanceId);
```

Comment:
only UpdateDbUsersAvatars should be in the data service. 
UploadUsersAvatarsAndGetFilenames should not be here. it is not related to the databse. Move it elsewhere and pass the fileNames as a parameter if needed